src/systems/filecoin_markets/storage_market/storage_client.id

```diff
@@ -1,18 +1,62 @@
 import ipld "github.com/filecoin-project/specs/libraries/ipld"
-import deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
-import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/storage_market/storage_deal"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import libp2p "github.com/filecoin-project/specs/libraries/libp2p"
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+
+type ClientDeal struct {
+    ProposalCid     ipld.CID
+    Proposal        deal.StorageDealProposal
+    State           StorageDealStatus
+    Miner           libp2p.PeerID
+    MinerWorker     addr.Address
+    DealID          deal.DealID
+    PublishMessage  ipld.CID
+}
+
+type ProposeStorageDealResult struct {
+    ProposalCid ipld.CID
+}
+
+type StorageProviderInfo struct {
+    Address     addr.Address  // actor address
+    Owner       addr.Address
+    Worker      addr.Address  // signs messages
+    SectorSize  uint64
+    PeerID      libp2p.PeerID
+}
 
 type StorageClient struct {
-    // generate PieceCID(CommP) from payload
-    // so that client can identify their deal
-    generatePieceCID(payloadCID ipld.CID) piece.PieceCID
+    // ListProviders queries chain state and returns active storage providers
+    ListProviders()        (providers [StorageProviderInfo], error)
+
+    // ListDeals lists on-chain deals associated with this client
+    ListDeals()            (deals [deal.StorageDeal], error)
+
+    // ListInProgressDeals lists deals that are in progress or rejected
+    // - but not on chain
+    ListInProgressDeals()  (deals [ClientDeal], error)
+
+    // GetInProgressDeal looks up an in progress deal by proposal CID
+    GetInProgressDeal(proposalCID ipld.CID) (ClientDeal, error)
+
+    // GetAsk returns the current ask for a storage provider
+    GetAsk(info StorageProviderInfo) (StorageAsk, error)
 
-    // Call by StorageProvider to pull data with GraphSync
-    PullPayload(payloadCID ipld.CID)
+    // ProposeStorageDeal initiates deal negotiation with a Storage Provider
+    ProposeStorageDeal(
+        info                StorageProviderInfo
+        payloadCid          ipld.CID
+        proposalExpiration  block.ChainEpoch
+        duration            block.ChainEpoch
+        price               actor.TokenAmount
+        collateral          actor.TokenAmount
+    ) (ProposeStorageDealResult, error)
 
-    // Call by StorageProvider to inform StorageDealProposalStatus
-    NotifyOfStorageDealProposalStatus(proposalCID deal.ProposalCID, status StorageDealStatus)
+    // GetPaymentEscrow returns the current funds available for deal payment
```

Comment:
can this be called by either party? If not, might be worth getting more specific with name, if so may be worth adding in a comment

Reply:
by nature of the EscrowTable existing in SMA, this should be public info and callable by any party.